### PR TITLE
Travis CI Test Updates / Python 3 Tests / Fix deprecated test methods.

### PR DESCRIPTION
**Pull request by XericZephyr**
Now the travis ci tests comes with python 3 tests.

This PR also fix some deprecated methods in tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +1,17 @@
-sudo: required
 dist: trusty
+sudo: false
 language: python
 python:
-- '2.7'
+  - "2.7.13"
+  - "3.6.1"
```
Comment (shengwa) [inline .travis.yml]: Do these versions need to be so specific as 3.6.x , 2.7.x instead of 3.6, 2.7?

**Review by shengwa — APPROVED** [review]
Good job! Don't forget to update the version. Speaking of version, it seems better if it is set inside the module smile as __version__.

LGTM

File: .travis.yml (hunk 1 above)
Comment (XericZephyr) [reply .travis.yml]: Done.

**Comment by XericZephyr** [comment]
@shengwa Hi, I have updated the python version in travis config with 2.7 and 3.6 instead. 

For the package version part, I am thinking of a bit more complex but more autonomous way to update the smile package version. If I cannot, I will change it in the next PR when we try to merge "master" to "production" branch. The deployment to PyPI on branch production is automated already.

**Comment by shengwa** [comment]
@XericZephyr I like the idea. Let me know if you need help.

**Pull request merged**